### PR TITLE
chore(d.ts): improve type safety for 'inputValue' attribute

### LINKED ISSUE #957: InputAttributes should accept (boolean | string | number) values
As you can see here: https://github.com/sweetalert2/sweetalert2/blob/138b5c87e84cb73e917c2efb81a5578b17067391/sweetalert2.d.ts#L632

... `inputAttributes` only accepts `string` values, however even in [examples](https://sweetalert2.github.io/), you can see `maxLength` _attribute_ being passed as `number`.

Also, I think it could accept also `boolean` allowing people to do something like:

```
inputAttributes: {
  autofocus: true
  // ...
}
```

<hr>

Also, I think the `inputValue` parameter should be `boolean | number | string` instead of `any`:

https://github.com/sweetalert2/sweetalert2/blob/138b5c87e84cb73e917c2efb81a5578b17067391/sweetalert2.d.ts#L602

### PR DESCRIPTION
**Pull request by rafaelss95**
Closes #957.

cc @toverux.

### CODE REVIEW

File: sweetalert2.d.ts [hunk 1]
```diff
@@ -550,7 +550,7 @@ declare module 'sweetalert2' {
          *
          * @default null
          */
-        preConfirm?: (inputValue: any) => SyncOrAsync<any | void>;
+        preConfirm?: (inputValue: string) => SyncOrAsync<any | void>;
```
Comment (toverux) [inline sweetalert2.d.ts]: This one is wrong: preConfirm can also take a boolean (it's actually the classic case, when there is no input).
I can find only one place where SweetAlert2 returns another thing than a string input value: https://github.com/sweetalert2/sweetalert2/blob/2b33d1906b5ec1c9963aa7540637f231398c782e/src/sweetalert2.js#L258

We can infer that the correct type is actually `true | string`.

But I don't like it. What if, one day, preConfirm can receive other data types? I propose to keep `any` for now.

Comment (rafaelss95) [reply sweetalert2.d.ts]: Done. I've removed this change. Take a look please :)

Comment (limonte) [reply sweetalert2.d.ts]: @toverux ping :)

**Comment by limonte** [comment]
I guess @toverux is in vacation-mode now :) I'll go ahead and merge this PR.

Thanks again @rafaelss95!

**Comment by limonte** [comment]
Released in https://github.com/sweetalert2/sweetalert2/releases/tag/v7.16.0

**Pull request merged**